app/src/main/java/org/hisp/dhis/android/dataentry/main/MainActivity.java

```diff
@@ -163,14 +164,31 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
 
         if (menuItem.getItemId() == R.id.drawer_item_home) {
             attachFragment(new HomeFragment());
-        } else {
-            attachFragment(new DummyFragment());
+        } else if (menuItem.getItemId() == R.id.drawer_item_log_out) {
+            showLogOutDialog();
         }
         drawerLayout.closeDrawers();
 
         return true;
     }
 
+    private void showLogOutDialog() {
+        new AlertDialog.Builder(this)
+                .setTitle(R.string.log_out_title)
+                .setPositiveButton(
+                        R.string.log_out_button,
+                        (dialog, which) -> logOut())
+                .setNegativeButton(android.R.string.cancel, null)
+                .setMessage(R.string.log_out_message).show();
+    }
+
+    private void logOut() {
+        ProgressDialog dialog = ProgressDialog.show(this, "Logging out", "Deleting all data");
+        mainPresenter.logOut();
```

Comment:
This is one instance where it actually makes sense to do IO on the main thread. The user has confirmed log out and the UI should be blocked until log out is complete.